featuretools/entityset/entity.py

```diff
@@ -249,10 +252,6 @@ def query_by_values(self, instance_vals, variable_id=None, columns=None,
         elif instance_vals.shape[0] == 0:
             df = self.df.head(0)
 
-        elif variable_id is None or variable_id == self.index:
-            df = self.df.reindex(instance_vals)
-            df.dropna(subset=[self.index], inplace=True)
-
         else:
             df = self.df[self.df[variable_id].isin(instance_vals)]
```

Comment:
I suspect the reason, we used reindex previously was because it is to select that way then `self.df[self.df[variable_id].isin(instance_vals)]`.

if that is the case, im wondering if there is a way to maintain the behavior, but get the speeds ups of using the index.

for example, there is the the isin method defined for pandas indices: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.isin.html

Reply:
here are some timings

```Python 3.6.4 (default, Dec 30 2017, 13:28:16) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pandas as pd 
   ...: import numpy as np 
   ...:  
   ...: nrows = 100000 
   ...: sample = 1000 
   ...: numeric_series = pd.Series(range(0, nrows)) 
   ...: numeric_index = pd.Index(numeric_series) 
   ...: numeric_sample = numeric_series.sample(sample).values 
   ...:  
   ...: str_series = numeric_series.astype(str) 
   ...: str_index = pd.Index(str_series) 
   ...: str_sample = str_series.sample(sample).values                                                                                                                                                                                         

In [2]: %timeit numeric_series[numeric_series.isin(numeric_sample)]                                                                                                                                                                           
2.18 ms ± 114 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [3]: %timeit numeric_index.reindex(numeric_sample)                                                                                                                                                                                         
87.2 µs ± 1.65 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [4]: %timeit numeric_index[numeric_index.isin(numeric_sample)]                                                                                                                                                                             
1.64 ms ± 13 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [5]: %timeit str_series[str_series.isin(str_sample)]                                                                                                                                                                                       
6.22 ms ± 126 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [6]: %timeit str_index.reindex(str_sample)                                                                                                                                                                                                 
241 µs ± 8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [7]: %timeit str_index[str_index.isin(str_sample)]                                                                                                                                                                                         
5.73 ms ± 221 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```